src/elements/autocomplete-grid/autocomplete-grid-button/autocomplete-grid-button.ts

```diff
@@ -0,0 +1,145 @@
+ */
+@customElement('sbb-autocomplete-grid-button')
+@hostAttributes({
+  role: 'button',
+  tabindex: null,
+  'data-button': '',
+})
+export class SbbAutocompleteGridButtonElement extends SbbDisabledMixin(
+  SbbNegativeMixin(SbbIconNameMixin(SbbActionBaseElement)),
+) {
+  public static override styles: CSSResultGroup = style;
+
+  /** Gets the SbbAutocompleteGridOptionElement on the same row of the button. */
+  public get option(): SbbAutocompleteGridOptionElement | null {
+    return (
+      this.closest('sbb-autocomplete-grid-row')?.querySelector('sbb-autocomplete-grid-option') ||
+      null
+    );
+  }
+
+  /** Whether the component must be set disabled due disabled attribute on sbb-optgroup. */
+  private _disabledFromGroup = false;
+
+  protected override isDisabledExternally(): boolean {
+    return this._disabledFromGroup ?? false;
+  }
+
+  /** MutationObserver on data attributes. */
+  private _optionAttributeObserver = new AgnosticMutationObserver((mutationsList) =>
+    this._onOptionAttributesChange(mutationsList),
+  );
+
+  /** Observe changes on data attributes and set the appropriate values. */
+  private _onOptionAttributesChange(mutationsList: MutationRecord[]): void {
+    for (const mutation of mutationsList) {
+      if (mutation.attributeName === 'data-group-disabled') {
+        this._disabledFromGroup = this.hasAttribute('data-group-disabled');
+        setOrRemoveAttribute(this, 'aria-disabled', `${this.disabled || this._disabledFromGroup}`);
+      }
+    }
+  }
+
+  public constructor() {
+    super();
+    new SbbSlotStateController(this);
+    if (!isServer) {
+      this.setupBaseEventHandlers();
+      this.addEventListener('click', this._handleButtonClick);
+    }
+  }
+
+  protected override renderTemplate(): TemplateResult {
+    return super.renderIconSlot();
+  }
+
+  public override connectedCallback(): void {
+    super.connectedCallback();
+    this.id ||= `sbb-autocomplete-grid-button-${++autocompleteButtonNextId}`;
+    const parentGroup = this.closest('sbb-autocomplete-grid-optgroup');
+    if (parentGroup) {
+      this._disabledFromGroup = parentGroup.disabled;
+      setOrRemoveAttribute(this, 'aria-disabled', `${this.disabled || this._disabledFromGroup}`);
+    }
+    this._optionAttributeObserver.observe(this, buttonObserverConfig);
+  }
+
+  public override willUpdate(changedProperties: PropertyValues<this>): void {
+    super.willUpdate(changedProperties);
+    if (changedProperties.has('disabled')) {
+      setOrRemoveAttribute(this, 'aria-disabled', `${this.disabled || this._disabledFromGroup}`);
+    }
+  }
+
+  public override disconnectedCallback(): void {
+    super.disconnectedCallback();
+    this._optionAttributeObserver.disconnect();
+  }
+
+  /**
+   * Used to dispatch a click event when users interact with the button via keyboard (the component does not receive focus).
+   */
+  public dispatchClick(event: KeyboardEvent): void {
```

Comment:
As usual, the click on the ac-grid-button emits a `click` event with the button component as a target;
I could move this method in the ac-grid, however this will cause to emit a forced `click` event on keyboard selection with the ac-grid as a target, even if the 'visual-focus' is on the button

Reply:
I'm not sure I fully understand.
My suggestion would be to move the dispatch click event logic to the grid (or somewhere re-usable?), but still dispatch the event on the grid button in the selection via keyboard scenario.
Could you elaborate again?

Reply:
Fixed calling directly the `click` on the ac-grid-button instance